sdk/python/kfp/compiler/_container_builder.py

```diff
@@ -31,6 +33,10 @@ def __init__(self, gcs_staging, namespace):
       raise ValueError('Error: {} should be a GCS path.'.format(gcs_staging))
     self._gcs_staging = gcs_staging
     self._namespace = namespace
+    if namespace is None:
+      with open(SERVICEACCOUNT_NAMESPACE, 'r') as f:
```

Comment:
Could you add a todo to handle not in cluster use case and assume that this code path will only be executed in cluster?

Reply:
Or you should handle if the file doesn't exist, and set default value for namespace.

Reply:
Good point. Done.